network/src/transport.rs

```diff
@@ -132,56 +143,56 @@ pub struct Connection<TSocket> {
     pub metadata: ConnectionMetadata,
 }
 
-fn identity_key_to_peer_id(
-    trusted_peers: &RwLock<HashMap<PeerId, NetworkPublicKeys>>,
-    remote_static_key: &[u8],
-) -> Option<PeerId> {
-    for (peer_id, public_keys) in trusted_peers.read().unwrap().iter() {
-        if public_keys.identity_public_key.to_bytes() == remote_static_key {
-            return Some(*peer_id);
-        }
+/// Convert a remote pubkey into a network `PeerId`.
+fn identity_pubkey_to_peer_id(
+    maybe_trusted_peers: Option<&Arc<RwLock<HashMap<PeerId, NetworkPublicKeys>>>>,
+    remote_pubkey: &x25519::PublicKey,
+) -> io::Result<PeerId> {
+    if let Some(trusted_peers) = maybe_trusted_peers {
+        // if mutual authentication, try to find peer with this identity pubkey.
+        find_peer_with_pubkey(&trusted_peers.read().unwrap(), remote_pubkey)
+            .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "Not a trusted peer"))
+            .map_err(|err| {
+                security_log(SecurityEvent::InvalidNetworkPeer)
+                    .error("UntrustedPeer")
+                    .data(&trusted_peers)
+                    .data(&remote_pubkey)
+                    .log();
```

Comment:
security_log is unmaintained, if you want to log this, go for it, but just use the regular logger

@mimoo are we deleting it and just going to depend on the regular structured logger?

Reply:
this is a good question. Do you have context on why there's a separate security_log? I'll start to look at security logs, intuition was that I could just use the current logging framework that we have with a specific flag to mark it for security

Reply:
@gedigi said that he introduced it at a time where there was no logging and we needed logging before launching test net. I think at this point in time, we should focus on having a single logging framework and there should be a component outside that grabs logs that it wants rather than having specific logging infra inside.

Reply:
tl;dr let's delete it

Reply:
so let's not delete it, and I'll replace these in another PR :)

Reply:
yeah, i'm not sure what/who uses security_log. I just preserved the current behaviour... should I remove this?

Reply:
again, do not remove it

Reply:
I guess I should have said delete and leave a TODO, my concern with just leaving it is that we don't really use it, so keeping it around just adds noise whereas a TODO is much clearer...

@gedigi @gedigi @gedigi 
I thought we had to say your name 3 times?